nio-impl/src/main/java/org/xnio/nio/WorkerThread.java

```diff
@@ -335,7 +335,9 @@ void handleReady(final int ops) {
                 }
             } catch (IOException e) {
                 selectorLog.tracef("ConnectHandle.handleReady Exception, %s", e);
-                futureResult.setException(e);
+                futureResult.setException(new IOException(String.format("%s - failed to connect to %s",
```

Comment:
@spyrkob please, add the exception message to org.xnio.nioLog

Reply:
sorry, I missed that. updated now